src/types/PaymentMethod.ts

```diff
@@ -0,0 +1,201 @@
+  | AfterpayClearpayParams
+  | KlarnaParams
+  // | WeChatPayParams
+  | BancontactParams;
+
+export type ConfirmParams = CreateParams;
+
+export type CreateOptions = {};
+
+export type ConfirmOptions = CreateOptions;
+
+export type ShippingDetails = BillingDetails;
+
+interface BaseParams {
+  billingDetails?: BillingDetails;
+}
+
+export type CardParams =
+  | (BaseParams & {
+      type: 'Card';
+      token?: string;
+      setupFutureUsage?: FutureUsage;
+    })
+  | (BaseParams & {
+      type: 'Card';
+      paymentMethodId: string;
+      cvc?: string;
+    });
+
+export interface IdealParams extends BaseParams {
+  type: 'Ideal';
+  bankName?: string;
+  setupFutureUsage?: FutureUsage;
+}
+
+export interface FPXParams {
+  type: 'Fpx';
+  testOfflineBank?: boolean;
+}
+
+export interface AlipayParams {
+  type: 'Alipay';
+}
+
+export interface OxxoParams extends Required<BaseParams> {
+  type: 'Oxxo';
+}
+
+export interface SofortParams extends BaseParams {
+  type: 'Sofort';
+  country: string;
+  setupFutureUsage?: FutureUsage;
+}
+export interface GrabPayParams extends BaseParams {
+  type: 'GrabPay';
+}
+
+export interface BancontactParams extends Required<BaseParams> {
+  type: 'Bancontact';
+  setupFutureUsage?: FutureUsage;
+}
+
+export interface SepaParams extends Required<BaseParams> {
+  type: 'SepaDebit';
+  iban: string;
+  setupFutureUsage?: FutureUsage;
+}
+
+export interface GiropayParams extends Required<BaseParams> {
+  type: 'Giropay';
+}
+
+export interface AfterpayClearpayParams extends Required<BaseParams> {
+  type: 'AfterpayClearpay';
+  shippingDetails: ShippingDetails;
+}
+
+export type KlarnaParams = {
+  type: 'Klarna';
+  billingDetails: Pick<Required<BillingDetails>, 'email'> & {
+    address: Pick<Required<Address>, 'country'>;
+  } & BillingDetails;
+};
+
+export interface EpsParams extends Required<BaseParams> {
+  type: 'Eps';
+}
+
+export interface P24Params extends Required<BaseParams> {
+  type: 'P24';
+}
+
+export interface WeChatPayParams extends BaseParams {
+  type: 'WeChatPay';
+  appId: string;
+}
+
+export interface AuBecsDebitParams {
+  type: 'AuBecsDebit';
+  formDetails: FormDetails;
+}
+
+export interface AuBecsDebitResult {
+  fingerprint?: string;
+  last4?: string;
+  bsbNumber?: string;
+}
+
+export interface BacsDebitResult {
+  sortCode?: string;
+  last4?: string;
+  fingerprint?: string;
+}
+
+export interface CardResult {
+  brand?: CardBrand;
+  country?: string;
+  expYear?: number;
+  expMonth?: number;
+  fingerprint?: string;
+  funding?: string;
+  last4?: string;
+}
+
+export interface FpxResult {
+  bank?: string;
+}
+
+export interface IdealResult {
+  bankIdentifierCode?: string;
+  bank?: string;
+}
+
+export interface SepaDebitResult {
+  bankCode?: string;
+  country?: string;
+  fingerprint?: string;
+  last4?: string;
+}
+
+export interface SofortResult {
+  country?: string;
+}
+
+export interface UpiResult {
+  vpa?: string;
+}
+
+export type Type =
```

Comment:
I generally recommend not naming things so generically, even if in context of this PaymenttMethod API source-file it's fine.

I worry about collisions and ambiguity when these files are imported in a single export index file. Thoughts?

Reply:
It's exported underneath the `PaymentMethod` module, please see the index file: https://github.com/stripe/stripe-react-native/pull/837/files#diff-a9ba9cbedd19c9f66d564d2e89912890209b98f0a7ef19187877d2587300e476R16